src/libponyc/codegen/genexe.c

```diff
@@ -279,9 +279,12 @@ static bool link_exe(compile_t* c, ast_t* program,
   size_t arch_len = arch - c->opt->triple;
   size_t ld_len = 128 + arch_len + strlen(file_exe) + strlen(file_o) +
     strlen(lib_args);
+  const char* linker = "ld";
+  if (c->opt->linker) {
+    linker = c->opt->linker;
+    ld_len += strlen(c->opt->linker);
+  }
```

Comment:
If the `linker` variable was set earlier in this function, it could be accounted for in the initial calculataion of `ld_len` above instead of adding to it here.

```c
const char* linker = c->opt->linker != NULL ? c->opt->linker :  "ld";
// ...
size_t ld_len = 128 + arch_len + strlen(linker) + strlen(file_exe) + strlen(file_o) + strlen(lib_args);
```

That seems cleaner than having this `if` statement here.

Unless I'm missing something?